src/app.ts

```diff
@@ -148,21 +164,23 @@ export class App extends Construct {
 
     fs.mkdirSync(this.outdir, { recursive: true });
 
+    const cache = new SynthRequestCache();
```

Comment:
How about we store a cache instance on the `App` instance? so that multiple calls to synth can reuse the same cache. I would move all global functions that accept an `App` to be a private method of `App` so it has easy access to the cache.

Reply:
The main reason I didn't do that as this could introduce a unintended side effect. Technically you could now call `synth` multiple times for the same app and modify it in between. Having a global store would break that. That's why I think it's the most simplest and safe way to have this per call. Most of the time where you call it only once, it wouldn't make a difference from a performance perspective. I prefer having it "stateless", otherwise we would need a method to clear the cache.

Reply:
Yeap, fair enough 👍